sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -746,6 +746,10 @@ export class EventHubReceiver extends LinkEntity {
             }
           },
           (err) => {
+            // restore events to the front of the internal queue.
```

Comment:
Why front of the queue?

Reply:
We take items from the front of the queue as we collect them, so we need to put them back in the reverse order to the front of the queue. Otherwise we could end up with our events being in the incorrect order if there are other events in the queue (unlikely in practice but why risk it! :smile:)

Reply:
Why shouldnt we give these events to the user instead of throwing the error?

Reply:
Ah, that's a different question and fair. We can definitely do that instead, but we'll have to duplicate some of our error handling logic to handle this.

For example, we'd have to look at the error that gets thrown from the `retry` function to see if it was `ReceiverDisconnectedError` so the caller of `receiveBatch` knows that it shouldn't attempt to call `receiveBatch` again (since this would result in our clients playing tug-of-war since they'd keep calling `receiveBatch` with an ownerLevel even if they don't have a claim to the partition.)

That said I can go down this route if you feel the benefit of getting the events to the user faster (versus waiting for the next receiveBatch call) is worth it!

Reply:
ok